PLUB/Sources/Views/Home/MainPage/ViewModel/CreateBoardViewModel.swift

```diff
@@ -78,32 +78,36 @@ final class CreateBoardViewModel: CreateBoardViewModelType {
     isSuccessCreateBoard = isSuccessCreatingBoard
       .asSignal(onErrorSignalWith: .empty())
     
-    onlyTextUploadButtonIsActivated = Observable.combineLatest(
-      writingTitle
-        .filter { $0 != Constants.titlePlaceholder },
-      writingContent
-        .filter { $0 != Constants.contentPlaceholder }
-    ) { ($0, $1) }
-      .map { !$0.isEmpty && !$1.isEmpty }
-      .asDriver(onErrorDriveWith: .empty())
-    
-    photoAndTextUploadButtonIsActivated = Observable.combineLatest(
-      writingTitle
-        .filter { $0 != Constants.titlePlaceholder },
-      writingContent
-        .filter { $0 != Constants.contentPlaceholder },
-      isSelectingImage
-    ) { ($0, $1, $2) }
-      .map { !$0.isEmpty && !$1.isEmpty && $2 }
-      .asDriver(onErrorDriveWith: .empty())
-    
-    onlyPhotoUploadButtonIsActivated = Observable.combineLatest(
-      writingTitle
-        .filter { $0 != Constants.titlePlaceholder },
-      isSelectingImage
-    ) { ($0, $1) }
-      .map { !$0.isEmpty && $1 }
-      .asDriver(onErrorDriveWith: .empty())
+    uploadButtonIsActivated = selectingPostType
+      .flatMap { type in
+      switch type {
+      case .photo:
+        return Observable.combineLatest(
+          writingTitle
+            .filter { $0 != Constants.titlePlaceholder },
+          isSelectingImage
+        ) { ($0, $1) }
+          .map { !$0.isEmpty && $1 }
+      case .text:
+        return Observable.combineLatest(
+          writingTitle
+            .filter { $0 != Constants.titlePlaceholder },
+          writingContent
+            .filter { $0 != Constants.contentPlaceholder }
+        ) { ($0, $1) }
+          .map { !$0.isEmpty && !$1.isEmpty }
+      case .photoAndText:
+        return Observable.combineLatest(
+          writingTitle
+            .filter { $0 != Constants.titlePlaceholder },
+          writingContent
+            .filter { $0 != Constants.contentPlaceholder },
+          isSelectingImage
+        ) { ($0, $1, $2) }
+          .map { !$0.isEmpty && !$1.isEmpty && $2 }
+      }
+    }
+    .asDriver(onErrorDriveWith: .empty())
     
```

Comment:
filter 해주는 부분 코드가 반복되는 것 같은데 아래와 같이 바꿔보면 어떨까요??

```suggestion
let titleNotEmpty = writingTitle
  .filter { $0 != Constants.titlePlaceholder }
  .map { !$0.isEmpty }

let contentNotEmpty = writingContent
  .filter { $0 != Constants.contentPlaceholder }
  .map { !$0.isEmpty }

let postType = selectingPostType
  .share()

let uploadButtonIsActivated = Observable.combineLatest(
  postType,
  titleNotEmpty,
  contentNotEmpty,
  isSelectingImage
)
.map { postType, title, content, image in
  switch postType {
  case .photo:
    return title && image
  case .text:
    return title && content
  case .photoAndText:
    return title && content && image
  }
}
.asDriver(onErrorDriveWith: .empty())

```

Reply:
확실히 코드가 더 깔끔해지네요. 수빈님 최고👍

Reply:
오오 수빈좌 미쳤다 👍

Reply:
근데 수빈좌 저거 안되네요 ㅇㅅ ㅇ 수정해야할꺼같아요